colibris/conf/settings.py

```diff
@@ -13,8 +13,7 @@
 
 MIDDLEWARE = [
     'colibris.middleware.errors.handle_errors_json',
-    'colibris.middleware.auth.handle_auth',
-    'colibris.middleware.schema.handle_schema_validation',
```

Comment:
@ccrisan, are we ok to do this? This might be a breaking change!

Reply:
@lucifurtun I'm ok with this as long as we don't lose this functionality and we allow an easy way for the automatic schema validation (and corresponding exceptions) to be used by simple views.

Reply:
What do you mean by "automatic schema validation". My opinion is that if the user decides to go with function based views, then it should be responsible to construct the schema explicitly inside that function.

Reply:
Before this PR, one would write something like this to e.g. update a resource:

    @request_schema(schemas.UserSchema())
    @response_schema(schemas.UserSchema())
    async def patch_user(request):
        user = models.User.create(**self.request['data'])
        result = schemas.UserSchema().dump(user)

        return web.json_response(result, status=201)

1. With `colibris.middleware.schema.handle_schema_validation`, `request['data']` would be curated, validated and ready to be used.
2. With `HTTPSchemaValidationError`, `colibris.middleware.errors.handle_errors_json` would prepare the expected "dedicated" errors in response.

How would these be accomplished now, with your proposal?